synapse/rest/media/v1/media_storage.py

```diff
@@ -114,12 +113,11 @@ def finish():
             with open(fname, "wb") as f:
                 yield f, fname, finish
         except Exception:
-            t, v, tb = sys.exc_info()
             try:
                 os.remove(fname)
             except Exception:
                 pass
-            raise t, v, tb
+            raise
```

Comment:
the idea here was to preserve the original traceback - IIRC a plain `raise` replaces the traceback with a new one.

Reply:
it does in fact preserve the traceback, you can try it out.

```python
In [3]: def x():
   ...:     def y():
   ...:         def z():
   ...:             raise AssertionError("asdajshd")
   ...:         z()
   ...:     y()
```

```python
In [5]: try:
   ...:     x()
   ...: except:
   ...:     raise
   ...: 
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-5-241c5ee12c20> in <module>()
      1 try:
----> 2     x()
      3 except:
      4     raise

<ipython-input-3-23356a912450> in x()
      4             raise AssertionError("asdajshd")
      5         z()
----> 6     y()

<ipython-input-3-23356a912450> in y()
      3         def z():
      4             raise AssertionError("asdajshd")
----> 5         z()
      6     y()

<ipython-input-3-23356a912450> in z()
      2     def y():
      3         def z():
----> 4             raise AssertionError("asdajshd")
      5         z()
      6     y()

AssertionError: asdajshd
```